linode_api4/objects/database.py

```diff
@@ -0,0 +1,405 @@
+    parent_id_name = 'database_id'
+
+    properties = {
+        'created': Property(is_datetime=True),
+        'id': Property(identifier=True),
+        'label': Property(),
+        'type': Property(),
+    }
+
+    def restore(self):
+        """
+        Restore a backup to a Managed Database on your Account.
+        """
+
+        return self._client.post('{}/restore'.format(self.api_endpoint), model=self)
+
+
+class MySQLDatabaseBackup(DatabaseBackup):
+    api_endpoint = '/databases/mysql/instances/{database_id}/backups/{id}'
+
+
+class MongoDBDatabaseBackup(DatabaseBackup):
+    api_endpoint = '/databases/mongodb/instances/{database_id}/backups/{id}'
+
+
+class PostgreSQLDatabaseBackup(DatabaseBackup):
+    api_endpoint = '/databases/postgresql/instances/{database_id}/backups/{id}'
+
+
+class MySQLDatabase(Base):
+    api_endpoint = '/databases/mysql/instances/{id}'
+
+    properties = {
+        'id': Property(identifier=True),
+        'label': Property(mutable=True, filterable=True),
+        'allow_list': Property(mutable=True),
+        'backups': Property(derived_class=MySQLDatabaseBackup),
+        'cluster_size': Property(),
+        'created': Property(is_datetime=True),
+        'encrypted': Property(),
+        'engine': Property(filterable=True),
+        'hosts': Property(),
+        'port': Property(),
+        'region': Property(filterable=True),
+        'replication_type': Property(),
+        'ssl_connection': Property(),
+        'status': Property(volatile=True, filterable=True),
+        'type': Property(filterable=True),
+        'updated': Property(volatile=True, is_datetime=True),
+        'updates': Property(mutable=True),
+        'version': Property(filterable=True),
+    }
+
+    @property
+    def credentials(self):
+        if not hasattr(self, '_credentials'):
+            resp = self._client.get('{}/credentials'.format(MySQLDatabase.api_endpoint), model=self)
+            self._set('_credentials', MappedObject(**resp))
+
+        return self._credentials
+
+    @property
+    def ssl(self):
+        if not hasattr(self, '_ssl'):
+            resp = self._client.get('{}/ssl'.format(MySQLDatabase.api_endpoint), model=self)
+            self._set('_ssl', MappedObject(**resp))
+
+        return self._ssl
+
+    def credentials_reset(self):
+        """
+        Reset the root password for a Managed MySQL Database.
+        """
+
+        self.invalidate()
+
+        return self._client.post('{}/credentials/reset'.format(MySQLDatabase.api_endpoint), model=self)
+
+    def patch(self):
+        """
+        Apply security patches and updates to the underlying operating system of the Managed MySQL Database.
+        """
+
+        self.invalidate()
+
+        return self._client.post('{}/patch'.format(MySQLDatabase.api_endpoint), model=self)
+
+    def backup_create(self, label, **kwargs):
+        """
+        Creates a snapshot backup of a Managed MySQL Database.
+        """
```

Comment:
Sphinx-style parameter/type definitions would be good to include here:

```suggestion
        :param label: The name for this backup
        :type label: str
        """
```